source/common/formatter/substitution_formatter.h

```diff
@@ -243,6 +244,247 @@ template <class FormatterContext> class FormatterBaseImpl : public FormatterBase
   std::vector<FormatterProviderBasePtr<FormatterContext>> providers_;
 };
 
+// Helper class to write value to output buffer in JSON style.
+class JsonStringSerializer {
+public:
+  using OutputBufferType = Json::StringOutput;
+  explicit JsonStringSerializer(std::string& output_buffer) : output_buffer_(output_buffer) {}
+
+  // Methods that be used to add JSON delimiter to output buffer.
+  void addMapBeginDelimiter() { output_buffer_.add("{"); }
```

Comment:
I'm not a huge fan of all the constants for these things, but why define them and not use them?

Reply:
I should remove that constants, sorry, I forget it. Thanks for pointing it out.